added minimum tls and corrected index (#2616)
cloudflare/cloudflare-docs: products/ssl/src/content/edge-certificates/additional-options/minimum-tls.md

---
order: 12
pcx-content-type: concept

> type: overview

Reviewer: ```suggestion
```

---
order: 12

> pcx-content-type: interim

Reviewer: Is there a reason for this to be an `interim` page? Why not a `concept` or ´reference`?

Author: Makes sense. I'll change to "concept".

type: overview
---

import MinimumTLSDefinition from "../../_partials/_minimum-tls-definition.md"

> # Minimum TLS Version

Reviewer: I think it's confusing to have a page title with `Minimum TLS Version` and then an `Overview`. Consider moving the info you have in this first part to somewhere below the sentence `In this case, it means that Cloudflare also accepts requests encrypted with all TLS versions beyond 1.0.`

<MinimumTLSDefinition/>

> For example, if TLS 1.1 is selected, visitors attempting to connect with TLS 1.0 will be rejected while visitors attempting to connect using TLS 1.1, 1.2, or 1.3 (if enabled) will be allowed to connect.

Reviewer: I'd break this sentence into two parts to make it easier to read. Example: `For example, if TLS 1.1 is selected, visitors attempting to connect with TLS 1.0 will be rejected. Visitors attempting to connect using TLS 1.1, 1.2, or 1.3 (if enabled) will be allowed to connect.`

Author: Agree

> To configure cipher suites, use the API. [Learn More](https://developers.cloudflare.com/ssl/ssl-tls/cipher-suites).

Reviewer: ```suggestion
You can use the API to [configure cipher suites](https://developers.cloudflare.com/ssl/ssl-tls/cipher-suites).
```

Reviewer: Hyperlinks should have context, so you should avoid links like `Learn more`

## Overview

Transport Layer Security (TLS) guarantees encrypted communications between a client and a web server via HTTPS. It replaces the now deprecated Secured Sockets Layer (SSL) protocol. When web traffic is encrypted with TLS, users see a green padlock in their browser window, near the URL box.

> You can manage the TLS version that your domain uses when proxied through Cloudflare by setting the **Minimum TLS Version** in the **SSL/TLS** tab of the Cloudflare dashboard. 

Reviewer: ```suggestion
You can manage the TLS version your domain uses when proxied through Cloudflare by setting the **Minimum TLS Version** in the **SSL/TLS** tab of the Cloudflare dashboard. 
```

Selecting a minimum version ensures that all subsequent, newer versions of the protocol are also supported.  TLS 1.0 is the version that Cloudflare sets by default for all customers using certificate-based encryption. In this case, it means that Cloudflare also accepts requests encrypted with all TLS versions beyond 1.0.

<Aside type="note">

Results on supported TLS versions from tools like SSL Labs are not accurate because such tools do not attempt a full TLS connection to your Cloudflare domain.

> </Aside>

Reviewer: I don't understand the intent of this message. What's the context?

Author: From what I understood there are tools, like [SSL Labs](https://www.ssllabs.com/ssltest/) that analyze SSL, but apparently it's not very accurate. It's not essencial information. I think it's probably better to remove.

> To properly test supported TLS versions, attempt a request to your Cloudflare domain while specifying a TLS version.  For example, use the curl command to test TLS 1.1 (replace `www.example.com` with your Cloudflare domain and hostname):

Reviewer: ```suggestion
To properly test supported TLS versions, attempt a request to your Cloudflare domain while specifying a TLS version.  For example, use a `curl` command to test TLS 1.1 (replace `www.example.com` with your Cloudflare domain and hostname):
```

```sh
$ curl https://www.example.com -svo /dev/null --tls-max 1.1
```

> If the TLS version is blocked by Cloudflare, the TLS handshake does not complete and instead returns errors such as:

Reviewer: ```suggestion
If the TLS version you are testing for is blocked by Cloudflare, the TLS handshake does not complete and returns an error:
```

Author: Do you agree with "If the TLS version you are testing is blocked by Cloudflare, the TLS handshake is not completed and returns an error:"?

Reviewer: Yes :)

**`* error:1400442E:SSL routines:CONNECT_CR_SRVR_HELLO:tlsv1 alert`**

For guidance on which TLS version to use, review the information outlined below.

## Understand TLS versions

A higher TLS version implies a stronger cryptographic standard.  TLS 1.2 includes fixes for known vulnerabilities found in previous versions.

As of June 2018, TLS 1.2 is the version required by the Payment Card Industry (PCI) Security Standards Council. Cloudflare recommends migrating to TLS 1.2 to comply with the PCI requirement.

TLS 1.3, which offers additional security and performance improvements, was approved by the Internet Engineering Task Force (IETF) in May 2018.

## Decide what version to use

Not all browser versions support TLS 1.2 and above. Depending on your particular business situation, this may present some limitations in using stronger encryption standards.

> Consider using TLS 1.0 or 1.1 for sites with a broad user base, particularly non-transactional sites. This way, you minimize the possibility that some clients might not be able to connect to your site securely.

Reviewer: ```suggestion
Consider using TLS 1.0 or 1.1 for sites with a broad user base, particularly non-transactional sites. This way, you minimize the possibility for some clients not being able to connect to your site securely.
```

Author: Do you agree with: "Consider using TLS 1.0 or 1.1 for sites with a broad user base, particularly non-transactional sites. This way, you minimize the possibility of some clients not being able to connect to your site securely."?

Reviewer: Yes, but I'd defer this one to native speakers

For a narrow user base and sites that run internal applications or business and productivity applications, Cloudflare recommends TLS 1.2. These sites might already have more stringent security requirements or might be subject to PCI compliance.  However, you also need to ensure that your users upgrade to a TLS 1.2 compliant browser.

> It is not recommended to set the minimum TLS to 1.3 as this will likely cause issues with search engine crawlers and certain browsers. It should only be set to TLS 1.3 where there is a specific use case.

Reviewer: ```suggestion
It is not recommended to set the minimum TLS to 1.3 as this will likely cause issues with search engine crawlers and certain browsers. You should only set your minimum TLS requirement to TLS 1.3 where you have specific use case needs.
```

Reviewer: Not sure about the last part of the sentence, but I feel it's lacking something to be complete - for example, which use cases are those that need TLS 1.3?

## Related resources

> ### More articles on PCI compliance

Reviewer: I think you should either remove the `More articles on PCI compliance` and `TLS and PCI compliance` H3 titles or change the later to an H4. As it is it's weird because you only have one article below `More articles on PCI compliance`, although it's plural.

Also, consider writing a description for `Transport Layer Security` and `PCI Security Standards Council` as you did for `PCI Compliance`

Author: How about removing the 2 subtitles and having the 3 links under "Related resources"?

Reviewer: Sounds good
